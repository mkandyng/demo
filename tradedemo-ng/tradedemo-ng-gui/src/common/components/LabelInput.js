import React from "react";

export default function LabelInput(props) {
    return (
      <p style={props.style}>
          <label>{props.label}:
               <input id={props.id}
                      type={props.type}
                      name={props.name}
                      step={props.step}
                      onChange={props.handleOnChange}
                      value={props.value} />
          </label>
      </p>
   )
}
