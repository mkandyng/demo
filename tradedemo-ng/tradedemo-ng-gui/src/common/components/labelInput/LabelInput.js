import React from "react";

export default function LabelInput(props) {
    const { id,
            label,
            style,
            type,
            name,
            step,
            value,
            handleOnChange } = props;
    return (
      <p style={style}>
          <label>{label}:
               <input id={id}
                      type={type}
                      name={name}
                      step={step}
                      onChange={handleOnChange}
                      value={value || ""} />
          </label>
      </p>
   )
}
