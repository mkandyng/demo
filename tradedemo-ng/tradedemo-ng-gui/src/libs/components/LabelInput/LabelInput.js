import React from "react";
import PropTypes from 'prop-types';

/**
 * [LabelInput Input box with Label wrapped with <p>]
 */
export default function LabelInput(props) {
    const { id,
            label,
            style,
            type,
            name,
            step,
            value,
            handleOnChange } = props;
    return (
      <p style={style}>
          <label>{label}:
               <input id={id}
                      type={type}
                      name={name}
                      step={step}
                      onChange={handleOnChange}
                      value={value || ""} />
          </label>
      </p>
   )
}

LabelInput.propTypes = {
    id: PropTypes.string,
    label: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    step: PropTypes.number,
    handleOnChange: PropTypes.func.isRequired
};
