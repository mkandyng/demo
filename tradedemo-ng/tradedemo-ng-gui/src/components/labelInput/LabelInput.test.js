import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LabelInput from "./LabelInput";

/**
 * This is unit test for the LabelInput widget
 * It leverage snapshot to test the presentation
 * and a functional test to check the onClick call
 */

describe("LabelInput", () => {
    const props = { id: "id",
                  label: "label",
                  style: "style",
                  type: "number",
                  name: "name",
                  step: 1,
                  value: 12,
                  handleOnChange: jest.fn() }

    it("should render component comparing with previous snapshot shallow markup", () => {
        // Given

        // When
        const component = shallow(<LabelInput {...props} />);

        // Then
        expect(toJson(component)).toMatchSnapshot();
    });

    it("should call handleOnClick prop with given event", () => {
        // Given
        const event = 1
        const component = shallow(<LabelInput {...props} />);

        // When
        component.find("input").simulate('change', event);

        // Then
        expect(props.handleOnChange).toBeCalledWith(event);
    });

});
