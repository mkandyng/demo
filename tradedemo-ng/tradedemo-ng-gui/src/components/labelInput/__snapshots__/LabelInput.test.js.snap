// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelInput should render component comparing with previous snapshot shallow markup 1`] = `
<p
  style="style"
>
  <label>
    label
    :
    <input
      id="id"
      name="name"
      onChange={[MockFunction]}
      step={1}
      type="number"
      value={12}
    />
  </label>
</p>
`;
